airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSource.java

```diff
@@ -494,12 +496,25 @@ public List<AutoCloseableIterator<AirbyteMessage>> getIncrementalIterators(final
       final ResultWithFailed<List<AirbyteStreamNameNamespacePair>> streamsUnderVacuum = streamsUnderVacuum(database,
           streamsCategorised.ctidStreams().streamsForCtidSync(),
           getQuoteString());
-      final List<ConfiguredAirbyteStream> finalListOfStreamsToBeSyncedViaCtid =
+
+      // Streams we failed to query for Vacuum - such as in the case of an unsupported postgres server
+      // are reclassified as xmin since we cannot guarantee that ctid will be possible.
+      reclassifyCategorisedCtidStreams(streamsCategorised, streamsUnderVacuum.failed());
```

Comment:
I see that this method is in CursorBasedCtidUtils. Should we move it because it will be used by xmin, ctid and cursor-based?

Reply:
I tried keeping this generic so I can implement in a single place, but couldn't find a way.
If you know how to make generic `<? extends classA, classB>` - couldn't find a workable solution